### PR TITLE
feat: workflow analytics code split and housekeeping

### LINKED ISSUE #31355: [Task]: Analytics Events
[Notion Doc](https://www.notion.so/appsmith/Workflows-Analytics-0c6bc3ace7fa4671a6d042cbef4d8ba0)

### PR DESCRIPTION
**Pull request by ayushpahwa**
## Description
This pr is for code splitting of analytics events for workflows project. It also does a pending code split for the jsPaneSagas which got missed during the audit logs feature release. [EE PR](https://github.com/appsmithorg/appsmith-ee/pull/3636) 

Changelog
- Added a context of parent for packages and workflows so the analytics events can have extra information.
- Updated code split for audit logs feature.

#### PR fixes following issue(s)
Fixes #31355 

#### Type of change

- New feature (non-breaking change which adds functionality)

#### Dev activity
- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] PR is being merged under a feature flag

#### QA activity:
- [ ] [Speedbreak features](https://github.com/appsmithorg/TestSmith/wiki/Guidelines-for-test-plans#speedbreakers-) have been covered
- [ ] Test plan covers all impacted features and [areas of interest](https://github.com/appsmithorg/TestSmith/wiki/Guidelines-for-test-plans#areas-of-interest-)
- [ ] Test plan has been peer reviewed by project stakeholders and other QA members
- [ ] Manually tested functionality on DP
- [ ] We had an implementation alignment call with stakeholders post QA Round 2
- [ ] Cypress test cases have been added and approved by SDET/manual QA
- [ ] Added `Test Plan Approved` label after Cypress tests were reviewed
- [ ] Added `Test Plan Approved` label after JUnit tests were reviewed

## Summary by CodeRabbit

- **New Features**
	- Enhanced analytics tracking for specific actions in the application, providing detailed insights into user behavior and action success rates.
	- Introduced context-aware data points for optimized analytics data collection, including parent context extraction from URLs.

### CODE REVIEW

File: app/client/src/ce/utils/analyticsHelpers.ts [hunk 1]
```diff
@@ -0,0 +1,7 @@
+/* eslint-disable @typescript-eslint/no-unused-vars */
+export const logMainJsActionExecution = (
+  actionId: string,
+  isSuccess: boolean,
+  collectionId: string,
+  isDirty: boolean,
+) => {};
```
[the remark this answers is not in the document]
Comment (ayushpahwa) [reply app/client/src/ce/utils/analyticsHelpers.ts:7]: This is a code split file for an ee feature. The file path is enough to clarify the same.

File: app/client/src/ce/actions/auditLogsAction.ts [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+  actionId: string;
+  pageId: string;
+  collectionId: string;
+  actionName: string;
+  pageName: string;
+}) => {
+  return {
+    type: "",
+  };
+};
```
[the remark this answers is not in the document]
Comment (ayushpahwa) [reply app/client/src/ce/actions/auditLogsAction.ts:15]: This is a code split file for an ee feature. The file path is enough to clarify the same.

**Review by ankitakinger — APPROVED** [review]
LGTM

File: app/client/src/utils/AnalyticsUtil.tsx [hunk 3]
```diff
@@ -18,6 +18,25 @@ declare global {
   }
 }
 
+const parentContextTypeTokens = ["pkg", "workflow"];
+
+// For app, function was returning app name due to the way app urls are structured
+// So this function will only return the parent context for pkg and workflow
+function getParentContextFromURL(location: Location) {
+  const pathSplit = location.pathname.split("/");
```
Comment (srix) [inline app/client/src/utils/AnalyticsUtil.tsx:26]: use docstring format in annotating function. please give an example path and what it will return

**Review by srix — CHANGES_REQUESTED** [review]

File: app/client/src/utils/AnalyticsUtil.tsx (hunk 3 above)
Comment (ayushpahwa) [reply app/client/src/utils/AnalyticsUtil.tsx:31]: updated

**Review by srix — APPROVED** [review]

**Pull request merged**